### PR TITLE
feat: add initial infrastructure setup for pillarbox monitoring

### PR DESCRIPTION
**Pull request by jboix**
## Description

This commit introduces the foundational Infrastructure as Code (IaC) setup for the Pillarbox Monitoring system to automate and simplify the deployment of these services.

- Configures Terraform for managing AWS resources with support for both dev and prod environments.
- Provides Docker Compose for local development and testing.

## Checklist

- [x] I have followed the project's style and contribution guidelines.
- [x] I have performed a self-review of my own changes.
- [x] I have made corresponding changes to the documentation.
- [x] I have added tests that prove my fix is effective or that my feature works.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+cmake-build-*/
+out/
+.idea_modules/
+
+# VisualStudioCode
+.vscode-test
+.vscode
+.history
+
+# macOs
```
Comment (defagos) [inline .gitignore:16]:
```suggestion
# macOS
```

😈

File: docs/README.md [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+**Accessing Grafana**: Grafana is accessible at [http://localhost:3000](http://localhost:3000).
+
+## Documentation
+
+The project is split into four main Terraform configurations:
+
+- [01-terraform-backend][terraform-backend]: This configuration sets up the Terraform backend and is
+  **only applicable in the prod environment**. The backend is used to store Terraform state
+  remotely, ensuring that multiple users and systems can work on infrastructure collaboratively.
+- [10-pillarbox-monitoring-route-53][route-53]: This handles the creation of the Route 53 DNS setup
```
Comment (defagos) [inline docs/README.md:112]: [Route 66](https://www.youtube.com/watch?v=AqhQSfFtOVE) would have been nicer. You're welcome.

File: pillarbox-monitoring-grafana/provisioning/dashboards/dashboard.yml [hunk 3]
```diff
@@ -0,0 +1,12 @@
+apiVersion: 1
+
+providers:
+  - name: 'Player QoS'
```
Comment (defagos) [inline pillarbox-monitoring-grafana/provisioning/dashboards/dashboard.yml:4]: Maybe

```suggestion
  - name: 'Player Monitoring'
```

would be nicer?

File: pillarbox-monitoring-terraform/20-pillarbox-monitoring-app/data_transfer_service.tf [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+          hostPort      = 8081
+          containerPort = 8081
+          protocol      = "tcp"
+        }
+      ]
+
+      # Environment variables for connecting to SSE and OpenSearch services
+      environment = [
+        {
+          name  = "PILLARBOX_QOS_SSE_URI"
```
Comment (defagos) [inline pillarbox-monitoring-terraform/20-pillarbox-monitoring-app/data_transfer_service.tf:68]:
```suggestion
          name  = "PILLARBOX_MONITORING_SSE_URI"
```

File: pillarbox-monitoring-terraform/20-pillarbox-monitoring-app/data_transfer_service.tf [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+          value = "http://${aws_service_discovery_service.dispatch_service_discovery.name}.${aws_service_discovery_private_dns_namespace.service_discovery_namespace.name}:8080/events"
+        },
+        {
+          name  = "PILLARBOX_QOS_OPENSEARCH_URI"
```
Comment (defagos) [inline pillarbox-monitoring-terraform/20-pillarbox-monitoring-app/data_transfer_service.tf:72]:
```suggestion
          name  = "PILLARBOX_MONITORING_OPENSEARCH_URI"
```

**Review by defagos — APPROVED** [review]
Cannot really provide a good feedback about the configuration itself (not proficient in this kind of stuff myself), I just reported a few things I noticed.

File: docker-compose.yml [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+
+  pillarbox-monitoring-transfer:
+    build: ./pillarbox-monitoring-transfer
+    image: srgssr/pillarbox-monitoring-transfer:1.0.0
+    container_name: pillarbox-monitoring-transfer
+    ports:
+      - "8081:8081"
+    environment:
+      - PILLARBOX_QOS_OPENSEARCH_URI=http://opensearch:9200
+      - PILLARBOX_QOS_SSE_URI=http://pillarbox-monitoring-dispatch:8080/events
```
Comment (MGaetan89) [inline docker-compose.yml:17]: Following @defagos's suggestions:
```suggestion
      - PILLARBOX_MONITORING_OPENSEARCH_URI=http://opensearch:9200
      - PILLARBOX_MONITORING_SSE_URI=http://pillarbox-monitoring-dispatch:8080/events
```

File: docs/CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+background can also submit changes directly via pull requests. Creating issues or pull requests
+requires you to own or [open](https://github.com/join) a GitHub account.
+
+If you are not sure about the likelihood of a change you propose to be accepted, please open an
+issue first. We can discuss it there, especially whether it is compatible with our product or not.
+This way you can avoid creating an entire pull request we will never be able to merge.
+
+Templates are available when you want to contribute:
+
+* [Issues](https://github.com/SRGSSR/pillarbox-monitoring-transfer/issues/new/choose): Please follow our issue
```
Comment (MGaetan89) [inline docs/CONTRIBUTING.md:48]:
```suggestion
* [Issues](https://github.com/SRGSSR/pillarbox-monitoring-infra/issues/new/choose): Please follow our issue
```

File: docs/CONTRIBUTING.md [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+  template. You can omit information which does not make sense but, in general, the more details you
+  can provide, the better. This ensures we can quickly reproduce the problem you are facing,
+  increasing the likelihood we can fix it.
+* [Pull requests](https://github.com/SRGSSR/pillarbox-monitoring-transfer/compare): Please follow our code
```
Comment (MGaetan89) [inline docs/CONTRIBUTING.md:52]:
```suggestion
* [Pull requests](https://github.com/SRGSSR/pillarbox-monitoring-infra/compare): Please follow our code
```

File: docs/README.md [hunk 9]
```diff
@@ -0,0 +1,166 @@
...
+- **Docker** and **Docker Compose**
+
+### Setup
+
+1. **Clone the Repository**
+
+   Clone the repository and initialize the submodules:
+
+   ```bash
+   git clone  --recurse-submodules <repository-url>
```
Comment (MGaetan89) [inline docs/README.md:25]:
```suggestion
   git clone  --recurse-submodules https://github.com/SRGSSR/pillarbox-monitoring-infra
```

File: docs/README.md [hunk 10]
```diff
@@ -0,0 +1,166 @@
...
+
+```bash
+git config core.hooksPath .githooks/
+```
+
+Refer to our [Contribution Guide](CONTRIBUTING.md) for more detailed information.
+
+## License
+
+This project is licensed under the [MIT License](/LICENSE).
```
Comment (MGaetan89) [inline docs/README.md:161]:
```suggestion
This project is licensed under the [MIT License](../LICENSE).
```

File: .editorconfig
Comment (MGaetan89) [inline .editorconfig:1]: What about adding a section of `*.tf` to ensure common formatting there too?

**Review by MGaetan89 — COMMENTED** [review]
A first set of comments, I still have some files to check

File: docs/README.md (hunk 2 above)
Comment (jboix) [reply docs/README.md:112]: You can contact AWS: https://aws.amazon.com/contact-us/ 😂

File: pillarbox-monitoring-terraform/20-pillarbox-monitoring-app/data_transfer_service.tf (hunk 4 above)
Comment (jboix) [reply pillarbox-monitoring-terraform/20-pillarbox-monitoring-app/data_transfer_service.tf:68]: I like the suggestion but unfortunately this one is tied to a configuration property for the transfer service: https://github.com/SRGSSR/pillarbox-monitoring-transfer/blob/089f51c828bc460491c1a3f185c23fa40c34427e/src/main/resources/application.yml#L5

File: pillarbox-monitoring-terraform/20-pillarbox-monitoring-app/data_transfer_service.tf (hunk 5 above)
Comment (jboix) [reply pillarbox-monitoring-terraform/20-pillarbox-monitoring-app/data_transfer_service.tf:72]: Same comment: https://github.com/SRGSSR/pillarbox-monitoring-infra/pull/1#discussion_r1794921805

File: docker-compose.yml (hunk 6 above)
Comment (jboix) [reply docker-compose.yml:17]: Same as https://github.com/SRGSSR/pillarbox-monitoring-infra/pull/1#discussion_r1794921805

**Comment by jboix** [comment]
@MGaetan89 and @defagos I've prepared another PR for the change from `qos` to `monitoring` here: https://github.com/SRGSSR/pillarbox-monitoring-transfer/pull/2 Take a look at that so that I can merge this one.

**Review by MGaetan89 — APPROVED** [review]

**Pull request merged**